config.go

```diff
@@ -83,44 +90,33 @@ func ReadPassword(filename string) (string, error) {
 	return conf.Password, nil
 }
 
+// WritePassword takes a filename and password. Locks the file (OS-level) and
+// writes the supplied password to the file
 func WritePassword(filename, password string) error {
 	conf := PrivateConfig{Password: password}
 	buf, err := yaml.Marshal(&conf)
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(filename, buf, 0600)
-}
 
-func Open(configFile string) (*CacophonyAPI, error) {
-	conf, err := ParseConfigFile(configFile)
-	if err != nil {
-		return nil, fmt.Errorf("configuration error: %v", err)
-	}
-	privConfigFilename := privConfigFilename(configFile)
-	password, err := ReadPassword(privConfigFilename)
+	fileLock := flock.New(filename)
```

Comment:
I'm pretty sure you don't want to use the config file itself as the lock file. Typically you use a file alongside the file you want to change as a lock file. The way it stands now, the `WriteFile` call below will overwrite the file, creating a new inode, which will make it look like the file is unlocked at that point. 

I suggest you make use a lock file either in /var/lock or by appending ".lock" to the password filename.